frontend/src/queries/nodes/DataTable/DataTable.tsx

```diff
@@ -334,8 +344,8 @@ export function DataTable({ query, setQuery, context, cachedResults }: DataTable
     ].filter((x) => !!x)
 
     const secondRowLeft = [
-        showReload ? canLoadNewData ? <AutoLoad /> : <Reload /> : null,
-        showElapsedTime ? <ElapsedTime /> : null,
+        isOnInsightPage(context) ? null : showReload ? canLoadNewData ? <AutoLoad /> : <Reload /> : null,
+        isOnInsightPage(context) ? null : showElapsedTime ? <ElapsedTime /> : null,
```

Comment:
I think the logic is upside down here. Think of the "frontend/src/queries" folder as a self-contained library.... and everything under "frontend/src/scenes" as "the posthog app".

Other than, yes, some query nodes being called "insight trends node", the data table itself shouldn't care about the context it's being called on. It shouldn't know or care that such a thing like "an insight page" even exists.

Is there any better way to get the same meaning across? Some things to ponder:

The queries have a bunch of parameters, and similar to the `/events` page, you can just add `showReload: false` onto new insights if you want to hide just this (I'd still show it thought)... so instead of moving the "if" into the data table, just have it at the scene level.

The less different yes/no state we have, the better. We're going to have a lot of different types of query nodes, and the complexity will skyrocket if there's a "insight mode", "dashboard mode", "full query mode", "inline mode", "superuser mode", etc. If we can have just "view" and "edit", or adding one, just "inline", "view" and "edit" that'd be rather contained. 

Right now, a query node with the minimum amount of params opens the "inline" view (e.g. `{kind: 'DataTableNode', source: {kind:'EventsNode'}}`). Adding "full:true" opens the edit view. There's no "view" mode.

Should we swap `full: true` to `mode: 'inline' | 'view' | 'edit'` or should this even be a separate parameter to the `<Query />` tag, not as part of the query itself?... 🤔

Reply:
Wasn't expecting https://github.com/PostHog/posthog/pull/14709 to grow so much... 

I was worried about the amount of `showX` this has and didn't want to add more. But this _is_ just a `showX` but with a different name.

Happy to switch it out... we can follow up by reducing the number of things you can toggle if need be

Reply:
![2023-03-13 17 32 25](https://user-images.githubusercontent.com/984817/224781847-9f616e62-3113-442f-a735-fc7bc95b2e75.gif)

If I'm not hiding the reload button then I don't need to pass that context in at all.

I think separately we need to make sure we can measure performance of these queries in the time-to-see data and then decide if we want to hide the reload button for the same reasons as with insights. But don't need to resolve that right now IMHO